Composer/packages/client/src/recoilModel/undo/undoHistory.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
Would be great to write unit tests for this piece as its a core piece of our functionality. Or is the history.test.tsx covers tests for all this functionality?

Reply:
Also this is super neat to use snapshots to navigate between undo redo states!

Reply:
I added the unit tests